examples/frame_prediction/js/entities/entities.js

```diff
@@ -0,0 +1,76 @@
+/************************************************************************************/
+/*                                                                                  */
+/*     Base entity that predicts its position based on the last update              */
+/*                                                                                  */
+/************************************************************************************/
+game.FramePredictionEntity = me.Entity.extend({
+   
+    init: function(x, y, settings) {
+        this._super(me.Entity, "init", [x, y , settings]);
+    },
+    
+    draw: function(renderer) {
+        // Predict the next position to draw at based on velocity
+        var updateInterval = (me.timer.getTime() - me.timer.lastUpdate) / ( (1000/me.sys.updatesPerSecond) );
```

Comment:
the second argument just basically contains the visible region to redraw. it used to be more extensively used by the former dirty rectangle algorithm (if you remember), but now we just basically pass the viewport (me.TMXLayer is most probably the one using it the most).

else for prediction purpose, why would the renderer not store the last time information when the last frame was rendered ? 

and as i was suggesting it somewhere else, what about we move the RaF code/callback and main loop into our renderer object ? it then can itself do all this kind of operations